src/CoreWCF.Kafka/src/CoreWCF/Channels/Helpers/KafkaCredentialTypeHelper.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace CoreWCF.Channels;
+
+internal static class KafkaCredentialTypeHelper
+{
+    public static bool IsDefined(KafkaCredentialType value)
+    {
+        return value == KafkaCredentialType.None
+               || value == KafkaCredentialType.SslKeyPairCertificate
```

Comment:
Are the values for KafkaCredentialType strictly incrementing by 1? E.g. the values are 0, 1, 2 etc. If so, a faster check is to do a bounds check on the integer value:  
```c#
return (int)value >= (int)KafkaCredentialType.None &&
       (int)value <= (int)KafkaCredentialType.SaslOAuthBearer;
```
Some of those casts might not be needed, but I put them in just to make sure.

Reply:
Resharper is suggesting this pattern matching saving one cast
```c#
(int)value is >= (int)KafkaCredentialType.None and <= (int)KafkaCredentialType.SaslOAuthBearer
```